bat-utils/lib/runtime-currency.js

```diff
@@ -1,649 +1,189 @@
-
-    if (validity.error) return runtime.captureException(validity.error, { extra: { trade: trade } })
-
-    flatlineP = false
-    if (dst === 'USDT') dst = 'USD'
-    if ((trade.e !== 'aggTrade') || (src === dst) || (config.allcoins.indexOf(src) === -1) ||
-        (config.allcoins.indexOf(dst) === -1)) {
-      return
-    }
-
-    if (!singleton.altrates[src]) singleton.altrates[src] = {}
-    singleton.altrates[src][dst] = trade.p
-
-    if (!singleton.altrates[dst]) singleton.altrates[dst] = {}
-    singleton.altrates[dst][src] = 1.0 / trade.p
-
-    Currency.prototype.altrates = singleton.altrates
-  }, () => { setTimeout(function () { monitor1a(symbol, true, config, runtime) }, 15 * msecs.second) })
+const mins5InSeconds = 5 * 60
+
+const knownRateKeys = [
+  'BTC',
+  'ETH',
+  'XRP',
+  'BCH',
+  'LTC',
+  'DASH',
+  'BTG',
+  'USD',
+  'EUR'
+]
+// satoshis, wei, etc.
+const decimals = {
+  BAT: 18,
+  BCH: 8,
+  BTC: 8,
+  ETC: 18,
+  ETH: 18,
+  LTC: 8,
+  NMC: 8,
+  PPC: 6,
+  XPM: 8,
+  ZEC: 8
 }
 
-const altcoins = {
-  _internal: {
-    f: async (altcoin, tickers, config, runtime) => {
-      const fiats = singleton.cache.get('fiats:' + altcoin)
-      const rates = {}
-      const unavailable = []
-      let now, rate, target
-
-      if (!fiats) return
-
-      fiats.forEach((fiat) => {
-        rate = singleton.cache.get('ticker:' + altcoin + fiat)
-        rates[fiat] = (rate && rate.last) || (unavailable.push(fiat) && undefined)
-      })
-      if (unavailable.length > 0) {
-        return runtime.captureException(altcoin + '.f fiat error: ' + unavailable.join(', ') + ' unavailable')
-      }
-
-      target = {}
-      target[altcoin] = rates
-      try { await rorschach(target, tickers, config, runtime) } catch (ex) {
-        now = underscore.now()
-        if (singleton.warnings > now) return
-
-        singleton.warnings = now + (15 * msecs.minute)
-        return runtime.captureException(ex)
+generateGlobal.knownRateKeys = knownRateKeys
+generateGlobal.decimals = decimals
+
+module.exports = generateGlobal
+
+Currency.prototype = {
+  decimals,
+  knownRateKeys,
+  access: async function (path) {
+    const context = this
+    const {
+      config,
+      cache
+    } = context
+    let {
+      url: currencyUrl,
+      access_token: accessToken
+    } = config
+    accessToken = accessToken || 'foobarfoobar'
+    const baseUrl = url.resolve(currencyUrl, '/v1/')
+    const endpoint = url.resolve(baseUrl, path)
+    const cached = cache.get(`currency:${endpoint}`)
+    if (cached) {
+      return cached
+    }
+    const options = {
+      useProxyP: true,
+      headers: {
+        authorization: `Bearer ${accessToken}`,
+        'content-type': 'application/json'
       }
     }
+    try {
+      const body = await braveHapi.wreck.get(endpoint, options)
+      const dataString = body.toString()
+      const data = JSON.parse(dataString)
+      const { payload } = data
+      cache.set(endpoint, payload)
```

Comment:
shouldn't this be `cache.set(currency:${endpoint}, payload)` to match the get above?